libsplinter/src/admin/rest_api/actix/ws_register_type.rs

```diff
@@ -134,28 +189,47 @@ impl AdminServiceEventSubscriber for WsAdminServiceEventSubscriber {
 #[derive(Debug, Serialize, Clone)]
 struct JsonAdminEvent {
     #[serde(serialize_with = "st_as_millis")]
-    timestamp: time::SystemTime,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    timestamp: Option<time::SystemTime>,
 
     #[serde(flatten)]
     event: AdminServiceEvent,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    event_id: Option<i64>,
 }
 
 impl From<(time::SystemTime, AdminServiceEvent)> for JsonAdminEvent {
     fn from(raw_evt: (time::SystemTime, AdminServiceEvent)) -> Self {
         Self {
-            timestamp: raw_evt.0,
+            timestamp: Some(raw_evt.0),
```

Comment:
These fields should actually both be set to the event id value, for backwards compatibility.

Reply:
This conversion is used by the `Mailbox`, so the timestamp is being used rather than the event ID

Reply:
But this is the JsonAdminEvent, which is what is sent out of the websocket to external subscribers

Reply:
The event ID isn't being produced when the `Mailbox` is being used, though.

Reply:
Perhaps this implementation should also be behind feature flags so this is obvious? The conversion happening below (`impl From<(i64, AdminServiceEvent)> for JsonAdminEvent`) is what is used when the event store is being used.

Reply:
Ok, I've updated this now to set the value of the `timestamp` to the current time (Since the field is typed as `SystemTime`), unless there is a different way you would suggest working around that type to set the `timestamp` to the event ID value.